docs/internals/optimizer.rst

```diff
@@ -985,6 +1018,9 @@ CommonSubexpressionEliminator, because SSA will make sure that the variables
 will not change and the CommonSubexpressionEliminator re-uses exactly the same
 variable if the value is known to be the same.
 
+Works best on code without literal arguments, which is the case when the code is in the
+:ref:`expression-split <expression-splitter>` form.
```

Comment:
I based this addition on the docstrings, which says this: https://github.com/ethereum/solidity/blob/fb11553a650f6e07299354a867fc80e1c1ba3c0b/libyul/optimiser/EqualStoreEliminator.h#L35

This basically means that that ExpressionSplitter is recommended before, right?

Reply:
it uses `isSimpleStore` of the `DataFlowAnalyzer`, which checks if the key and value arguments are identifiers, so indeed literals as arguments would fly under the radar and expression split form is recommended. This is also a place where destroying the strict expression split form would hurt (f ex in load resolver or literal rematerializer).